backend.js

```diff
@@ -169,22 +227,16 @@ Backend.prototype.getTile = function(z, x, y, callback) {
             headers = head || {};
             return makevtile(body, 'pbf');
         // Image sources do not allow overzooming (yet).
-        } else if (bz < z && headers['x-vector-backend-object'] !== 'fillzoom') {
+        } else if (opts.z < z && headers['x-vector-backend-object'] !== 'fillzoom') {
             headers['x-vector-backend-object'] = 'empty';
             return makevtile();
         } else {
             size = body.length;
             headers = head || {};
             return makevtile(body);
         }
-    };
-
-    sourceGet.scale = scale;
-    sourceGet.legacy = legacy;
-    sourceGet.upgrade = upgrade;
-    source.getTile(bz, bx, by, sourceGet);
```

Comment:
Yeah that first change changes a lot on how this is working. The main part is, that we do not `getTile` instead we introduce `getAsync` and use it as a wrapper for `getTitle`.

One difference between the two seems to be how they pass on some of the option parameters. `getTile`  uses properties added to the callback, `getAsync` uses an option object.

The master branch code uses the properties on the callback and has similar assumptions on how this is done on the `source`.

The customized code only uses the options object.

The `source.getTile` call with the callback function `sourceGet` is replaced with a `source.getAsync(opts)` call in the custom changes.

This is where we need to make sure that `scale` `legacy` and `upgrade` are passed through. And it looks like that this is happening, because these values come from the surrounding `_getTile()`. From there you can follow them to `getAsync()` and to `getTile` where the `opts` are created by using the `callback` values from any other call.

I hope that helps. At least that's how I came to the conclusion that it's all fine :-)